### ISSUE TITLE
JupyterHub server within Docker container

### ISSUE THREAD

**Issue by TiemenSch** [minrk/jupyterhub-demo#1]
Hi minrk,

I am experimenting with your setup again, and I was wondering if it would be possible to run the Hub's server alongside the singleuser containers using the sock approach:

```bash
docker run -v /var/run/docker.sock:/var/run/docker.sock -ti jupyterhub:latest
```

Where jupyterhub:latest would be the Docker image to make. 

If I understand correctly, the hub container should then be able to access its parent docker daemon. I was hoping that would make it able to spawn the singleuser containers in the same shared parent.

Any ideas how we could make that work?

**Comment by TiemenSch** [comment]
I tried to setup a repo following this idea, but it's not in a working condition yet. https://github.com/TiemenSch/jupyterhub-docker

**Comment by minrk** [comment]
Yes, you can definitely do this. You will need to install the `dockerspawner` package in the image, so you won't want to use the `jupyterhub` base image, instead your own Dockerfile that includes `pip3 install dockerspawner`.
